packages/nextra-theme-blog/src/basic-layout.tsx

```diff
@@ -0,0 +1,27 @@
+import Head from 'next/head'
+import React, { ReactNode, useEffect } from 'react'
+import { useBlogContext } from './blog-context'
+import { HeadingContext } from './mdx-theme'
+import { getTitle } from './utils/title'
+
+export const BasicLayout = ({ children }: { children: ReactNode }) => {
+  const { config, opts } = useBlogContext()
+  const { title, pageTitle } = getTitle({ config, opts })
+  const ref = React.useRef<HTMLHeadingElement>(null)
+  const { jsxH1 } = opts
+  const portalHeading = jsxH1
+  return (
+    <article className="container prose prose-sm dark:prose-dark md:prose">
+      <Head>
+        <title>{title}</title>
+        {config.head?.({ title, meta: opts.meta })}
+      </Head>
+      <HeadingContext.Provider value={ref}>
+        {portalHeading ? <h1 ref={ref}></h1> : null}
```

Comment:
```suggestion
        {portalHeading ? <h1 ref={ref} /> : null}
```

Reply:
You can also use ternary to render either h1 with/without ref

Reply:
I can't use ternary here since the content of h1 is from React.createportal. It needs to be removed with h1 node.

Reply:
it's will be removed https://github.com/shuding/nextra/pull/602#discussion_r932204813